src/diet.ts

```diff
@@ -738,32 +754,71 @@ export function consumeDiet(diet: Diet<Note>, name: DietName): void {
           consumeSafe(countToConsume, menuItem.item);
-        } else if ([Mayo.flex, Mayo.zapine].includes(menuItem.item)) {
+        };
+      };
+
+      const speakeasyDrinks: [Item, ItemAction][] = Object.keys(getClanLounge())
+        .map((s) => toItem(s))
+        .filter((i) => i.inebriety > 0)
+        .map((drink) => [
+          drink,
+          (countToConsume: number, menuItem: MenuItem<Note>) => {
+            cliExecute(`drink ${countToConsume} ${menuItem.item}`);
+          },
+        ]);
+      const mayoActions: [Item, ItemAction][] = Object.values(Mayo).map((i) => [
+        i,
+        (countToConsume: number, menuItem: MenuItem<Note>) => {
           MayoClinic.setMayoMinder(menuItem.item, countToConsume);
-        } else if (menuItem.item === $item`pocket wish`) {
-          acquire(1, $item`pocket wish`, 60000);
-          cliExecute(`genie effect ${menuItem.effect}`);
-        } else if (menuItem.item === $item`Rethinking Candy`) {
-          synthesize(countToConsume, $effect`Synthesis: Greed`);
-        } else if (getClanLounge()[`${menuItem.item}`] && itemType(menuItem.item) === "booze") {
-          cliExecute(`drink ${menuItem.item}`);
-        } else if (menuItem.item === $item`campfire hot dog`) {
-          // mafia does not support retrieveItem on campfire hot dog because it does not work on stick of firewood
-          if (!have($item`stick of firewood`)) {
-            buy(1, $item`stick of firewood`, garboValue($item`stick of firewood`));
-          }
-          consumeSafe(countToConsume, menuItem.item);
-        } else if (menuItem.item !== $item`Special Seasoning`) {
+        },
+      ]);
+
+      const itemActions = new Map<Item, ItemAction | "skip">([
+        [saladFork, elementalResistAction($element`hot`)],
+        [frostyMug, elementalResistAction($element`cold`)],
+        [
+          $item`pocket wish`,
+          (countToConsume: number, menuItem: MenuItem<Note>) =>
+            acquire(countToConsume, $item`pocket wish`, 60000) &&
+            cliExecute(`genie effect ${menuItem.effect}`),
+        ],
+        [
+          $item`campfire hot dog`,
+          (countToConsume: number, menuItem: MenuItem<Note>) => {
+            // mafia does not support retrieveItem on campfire hot dog because it does not work on stick of firewood
+            if (!have($item`stick of firewood`)) {
+              buy(1, $item`stick of firewood`, ingredientCost($item`stick of firewood`));
+            }
+            consumeSafe(countToConsume, menuItem.item);
+          },
+        ],
+        [$item`Special Seasoning`, "skip"],
+        [
+          $item`Rethinking Candy`,
+          (countToConsume: number, menuItem: MenuItem<Note>) =>
+            synthesize(countToConsume, menuItem.effect ?? $effect`Synthesis: Greed`),
+        ],
+        ...mayoActions,
+        ...speakeasyDrinks,
+      ]);
+
+      for (const menuItem of menuItems) {
+        const itemAction = itemActions.get(menuItem.item);
```

Comment:
I think you can just do `itemActions.get(menuItem.item)?.()` and have it be `| null` rather than `| "skip"`.

Reply:
No, because i use falsey to see if there is no action, right? 
I guess I can rework the code to use null instead, but it requires an additional conditional or that ? you have